src/credssp/mod.rs

```diff
@@ -437,13 +437,12 @@ impl<C: CredentialsProxy<AuthenticationData = AuthIdentity> + Send> CredSspServe
         })
     }
 
-    #[allow(clippy::result_large_err)]
     #[instrument(fields(state = ?self.state), skip_all)]
     pub fn process(
         &mut self,
         ts_request: TsRequest,
-    ) -> Generator<NetworkRequest, crate::Result<Vec<u8>>, Result<ServerState, ServerError>> {
-        Generator::<NetworkRequest, crate::Result<Vec<u8>>, Result<ServerState, ServerError>>::new(
+    ) -> Generator<NetworkRequest, crate::Result<Vec<u8>>, Result<ServerState, Box<ServerError>>> {
+        Generator::<NetworkRequest, crate::Result<Vec<u8>>, Result<ServerState, Box<ServerError>>>::new(
```

Comment:
question: Why did you box the error? Is it because it makes the return type too big otherwise?

Reply:
🤔 I did not think about it. Good idea! done